### PR TITLE
bpf: suppress unneeded CT entry + minor cleanup

### PR DESCRIPTION
**Pull request by lkpdn**
AFAIK if `ct_state->addr` is not empty and it turns out that the special loopback handling is unneeded, extra CT entry whose saddr and daddr are both the same selected target address is genarated, which is meaningless. This patch suppresses it, plus some minor cleanup.

If I'm missing something, sorry to bother you but please let me know why it's needed.

```release-note
Suppress needless extra CT generation in non-loopback service case.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7489)

### CODE REVIEW

**Comment by lkpdn** [comment]
rebased to fix incorrect SHA

**Comment by brb** [comment]
test-me-please

File: bpf/lib/conntrack.h [hunk 1]
```diff
@@ -718,27 +718,24 @@ static inline int __inline__ ct_create4(void *map, struct ipv4_ct_tuple *tuple,
 	if (map_update_elem(map, tuple, &entry, 0) < 0)
 		return DROP_CT_CREATE_FAILED;
 
-	if (ct_state->addr) {
+	if (ct_state->addr && ct_state->loopback) {
```
Comment (joestringer) [inline bpf/lib/conntrack.h:721]: This seems good to me.

I have just one thought, do we still need to create two entries for the loopback case?

**Review by joestringer — APPROVED** [review]
LGTM, I have just one question below but I don't think it necessarily affects this change.

This will also need a rebase for the bpf.sha after the other PR went in.

**Comment by joestringer** [comment]
Green CI build.

File: bpf/lib/conntrack.h (hunk 1 above)
Comment (lkpdn) [reply bpf/lib/conntrack.h:721]: We need two entries in loopback case. Non-loopback case had used it in a completely different intention which this patch removes. OTOH we might have some room to refactor to make things simple around this `ct_state` use.

**Comment by lkpdn** [comment]
Thanks for the review.
Rebased + SHA fixed.

**Pull request merged**